### PR TITLE
decrease test pv controller resync period to try to deflake api update conflicts

### PR DESCRIPTION
**Pull request by msau42**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
Trying to reduce flakes from https://github.com/kubernetes/kubernetes/issues/85320 by increasing the frequency of retrying API object updates in the integration test PV controller.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by msau42** [comment]
/hold
i'm going to analyze the timeline a little more closely

**Comment by msau42** [comment]
After the detailed investigation in https://github.com/kubernetes/kubernetes/issues/85320, I think this is still the quickest/simplest fix for the flaky test.  We can consider a real fix in the controller separately.

File: test/integration/volumescheduling/volume_binding_test.go [hunk 1]
```diff
@@ -926,8 +926,10 @@ func setupCluster(t *testing.T, nsName string, numberOfNodes int, resyncPeriod t
 
 func initPVController(context *testContext, provisionDelaySeconds int) (*persistentvolume.PersistentVolumeController, informers.SharedInformerFactory, error) {
 	clientset := context.clientSet
-	// Informers factory for controllers, we disable resync period for testing.
-	informerFactory := informers.NewSharedInformerFactory(clientset, 0)
+	// Informers factory for controllers
+	// Use a frequent resync period to retry API update conflicts due to
+	// https://github.com/kubernetes/kubernetes/issues/85320
+	informerFactory := informers.NewSharedInformerFactory(clientset, 5)
```
Comment (liggitt) [inline test/integration/volumescheduling/volume_binding_test.go]: 5 nanoseconds? the default is 12 hours... disabling completely (as we currently do) is more representative of how the controllers will actually behave. I would worry that an abnormally short resync would mask real problems in how the controller requeues/reattempts things.

Comment (msau42) [reply test/integration/volumescheduling/volume_binding_test.go]: good catch I meant 5 seconds and the controller resync, not informer resync

File: test/integration/volumescheduling/volume_binding_test.go [hunk 2]
```diff
@@ -946,10 +945,11 @@ func initPVController(context *testContext, provisionDelaySeconds int) (*persist
...
 	plugins := []volume.VolumePlugin{plugin}
 
-	controllerOptions := persistentvolumeoptions.NewPersistentVolumeControllerOptions()
 	params := persistentvolume.ControllerParameters{
-		KubeClient:                clientset,
-		SyncPeriod:                controllerOptions.PVClaimBinderSyncPeriod,
+		KubeClient: clientset,
+		// Use a frequent resync period to retry API update conflicts due to
+		// https://github.com/kubernetes/kubernetes/issues/85320
+		SyncPeriod:                5 * time.Second,
```
Comment (liggitt) [inline test/integration/volumescheduling/volume_binding_test.go]: this takes the attempts from 45 seconds to 15, assuming 3 retries are needed?

Comment (liggitt) [reply test/integration/volumescheduling/volume_binding_test.go]: Is it still worth waiting for the PVs to be marked available before creating the PVCs in TestVolumeBinding?

Comment (msau42) [reply test/integration/volumescheduling/volume_binding_test.go]: 10 seconds, assuming 1st conflict occurs at time 0. test timeout is 30 seconds. I could check for PVs to be available too, which would take the worse-case wait down to 5 seconds

Comment (msau42) [reply test/integration/volumescheduling/volume_binding_test.go]: added one more commit to wait for PVs to be available

File: test/integration/volumescheduling/volume_binding_test.go [hunk 3]
```diff
@@ -1180,6 +1194,20 @@ func validatePVPhase(t *testing.T, client clientset.Interface, pvName string, ph
 	}
 }
 
+func waitForPVPhase(client clientset.Interface, pvName string, phase v1.PersistentVolumePhase) error {
+	return wait.Poll(time.Second, 1*time.Second, func() (bool, error) {
```
Comment (liggitt) [inline test/integration/volumescheduling/volume_binding_test.go]: use PollImmediate to avoid delaying one second on every call

Comment (liggitt) [reply test/integration/volumescheduling/volume_binding_test.go]: doesn't the second arg mean this times out after one attempt?

Comment (msau42) [reply test/integration/volumescheduling/volume_binding_test.go]: fixed

**Comment by liggitt** [comment]
Once CI is green, go ahead and pick to release-1.17

**Pull request merged**